packages/lib/src/interactions/hooks.ts

```diff
@@ -62,15 +62,15 @@ export function useZoomOnSelection() {
 }
 
 function useWheelCapture() {
-  const canvasWrapper = useCanvasWrapper();
+  const canvasArea = useCanvasArea();
```

Comment:
`useCanvasArea` still relies on the DOM structure. Would it be possible to leverage the work of this PR to avoid this ?

For example, putting the ref on `canvasArea` instead of `visCanvas` and exposing `visCanvas` as `canvasArea.parentElement` ? Or the other way round ?

Reply:
I remember now why I was retrieving `visCanvas` via `useThree` instead of injecting it through context. For every state we add to `VisCanvas` to store a DOM element in the context, we basically add a new re-render on initial mount.

R3F probably re-renders once when it gets a reference to the `canvas` element. By then, we know that the canvas' parent elements are also in the DOM, so by accessing them with `parentElement`, we save a lot of processing.

I do agree that this is not very elegant, though. I'll have a bit more of a think.

Reply:
What's not super elegant is to have custom hooks for accessing specific parents of the `canvas`. Maybe I can still call `gl.domElement.parentElement[.parentElement]` to retrieve the elements, but do so in `VisCanvasProvider` instead of in custom hooks. No `useState`, no extra renders, no custom hooks.

Reply:
> Maybe I can still call gl.domElement.parentElement[.parentElement] to retrieve the elements, but do so in VisCanvasProvider instead of in custom hooks. No useState, no extra renders, no custom hooks.

Yeah, sounds good to me :+1:

Reply:
> Hm. Is it mandatory to use a state rather than a ref ?

Yeah, because the `ref` is `null` until React actually renders the DOM element. It'd be fine if we used the reference to the DOM element inside a `useEffect`, but we use it during render (with `createPortal(<></>, targetElem)`)